### PR TITLE
maint: changes for 2025.0 compiler

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
### Description:
**Maint: Changes for 2025.0 Compiler Compatibility**

#### Summary:
This PR updates the oneDAL codebase for compatibility with the upcoming 2025.0 compiler version. It includes fixes and modifications across multiple files to address issues related to the new compiler requirements.

#### Key Changes:
1. **Compute Kernel CSR Implementation:**
   - Updated `compute_kernel_csr_impl_dpc.cpp` for conditional compilation of `sycl::fmin` and `sycl::fmax`.

2. **Compute Kernel Dense Implementation:**
   - Modified `compute_kernel_dense_impl_dpc.cpp` for proper handling of `sycl::sqrt`.

3. **Test Adjustments:**
   - Updated test files (`badargs.cpp`, `overflow.cpp`) for algorithms like `covariance`, `dbscan`, `kmeans`, `linear_kernel`, and `pca`.

4. **Backend and Primitives:**
   - Adjusted backend and primitive test files for consistency with new macro definitions and enhanced stability.

### CODE REVIEW

File: cpp/oneapi/dal/test/archives.cpp [hunk 1]
```diff
@@ -18,17 +18,24 @@
...
 
-TEMPLATE_TEST("can write to binary_ouput_archive",
-              "[binary_ouput_archive]",
-              float,
-              double,
-              std::int32_t) {
+template <typename T>
+class archives_test : te::algo_fixture {
+public:
+};
```
Comment (ethanglaser) [inline cpp/oneapi/dal/test/archives.cpp:31]: No logic in the class? Should algo_fixture just be used?

File: cpp/oneapi/dal/backend/primitives/test/ndarray.cpp
Comment (ethanglaser) [inline cpp/oneapi/dal/backend/primitives/test/ndarray.cpp:1]: Please uncomment before merge or provide more formal method of skipping

**Review by ethanglaser — COMMENTED** [review]
Nice work on this

File: cpp/oneapi/dal/test/archives.cpp (hunk 1 above)
Comment (Alexandr-Solovev) [reply cpp/oneapi/dal/test/archives.cpp:31]: YEs, its necessary just to add a class with template parameter for TEMPLATE_LIST_TEST_M macros usage

Comment (Alexandr-Solovev) [reply cpp/oneapi/dal/test/archives.cpp:31]: Upd, it was replaced with more suitable macro

**Comment by Alexandr-Solovev** [comment]
CEV job: https://intel-ci.intel.com/ef5578fd-a8f6-f1a3-8e05-a4bf010d0e2e

**Review by icfaust — APPROVED** [review]
Please make a ticket for the observed windows sycl issue, otherwise CIs are green, and the changed to tests/ basic_statistics / macro.hpp are straightforward.  Just make sure someone else just glances at it other than me (verbal approval sufficient).

**Review by samir-nasibli — APPROVED** [review]

**Comment by ethanglaser** [comment]
Looking good - can we ensure than CI job with nightly (not CEV) is running smoothly?

**Comment by Alexandr-Solovev** [comment]
Hi @ethanglaser , nope, with this pr CEV/MEV will be red, but due to it necessary https://github.com/intel/scikit-learn-intelex/pull/2000, I guess we can merge it in the same time as we merge your prs with cl removals

**Comment by ethanglaser** [comment]
> Hi @ethanglaser , nope, with this pr CEV/MEV will be red, but due to it necessary [intel/scikit-learn-intelex#2000](https://github.com/intel/scikit-learn-intelex/pull/2000), I guess we can merge it in the same time as we merge your prs with cl removals

Yes I mean regular nightly scope (not CEV)

**Comment by homksei** [comment]
Jobs with ICX as a build compiler: 
- Linux:  [intel-ci.intel.com/ef65521e-d0d2-f199-b313-a4bf010d0e2e](https://intel-ci.intel.com/ef65521e-d0d2-f199-b313-a4bf010d0e2e)
- Windows: [intel-ci.intel.com/ef65e568-6c9f-f164-a693-a4bf010d0e2e](https://intel-ci.intel.com/ef65e568-6c9f-f164-a693-a4bf010d0e2e)

(builds passed)

**Review by homksei — APPROVED** [review]

**Pull request merged**